udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,36 +44,48 @@ private String arrayListToJson(ArrayList<String> inputList) {
   }
 
   /**
-  * Obtains comments json to display upon request.
-  */
+   * Obtains comments json to display upon request.
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
+    Query query = new Query("Comms").addSort("timestamp", SortDirection.DESCENDING);
+
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> comments = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+      String commentText = (String) entity.getProperty("text");
+      comments.add(commentText);
+    }
+
     String outputJson = arrayListToJson(comments);
+
     response.getWriter().println(outputJson);
   }
 
   /**
-  * Writes comment data to ArrayList data
-  */
+   * Writes comment data to ArrayList data
+   */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
     String text = getParameter(request, "text-input", "");
-    boolean eraser = Boolean.parseBoolean(getParameter(request, "erase", "false"));
-    comments.add(text);
+    long timestamp = System.currentTimeMillis();
 
-    if (eraser) {
-      comments = new ArrayList<String>();
-    }
+    Entity taskEntity = new Entity("Comms");
```

Comment:
I'm curious how you determined that "Comms" should be used to initialize the Entity. Could you add a comment explaining the meaning of that string?

Just a guess, is it short for "Comments"? Could the entity be named "Comments" instead to avoid confusion?

Reply:
Done